### ISSUE TITLE
no-floating-promises not being reported in VSCode (Works via CLI)

### ISSUE THREAD

**Issue by Recodify** [palantir/tslint#4245]
### Bug Report

- __TSLint version__:  5.11.0
- __TypeScript version__: 3.1.3
- __Running TSLint via__:  VSCode 
- __VSCode version__: 1.28.2
- __TSLint extension version__: 1.0.40 

#### TypeScript code being linted

```ts
class MyClass {
   
    async test() : Promise<string> {
        return "new";
    }

    test2() {
       // not awaiting this so should error.
       this.test();
    }
}
```

with `tslint.json` configuration:

```json
{    
    "rules": {
        "no-floating-promises": true,
        "semicolon": true
    }
}
```

#### Actual behavior

When running via CLI, works as expected with error: ERROR: /home/me/code/scratch/app.ts[10, 8]: Promises must be handled appropriately

However via VSCode, no problems are reported. Other rules work OK.

![image](https://user-images.githubusercontent.com/5644228/47389027-8faea680-d70b-11e8-916d-b4ff1c668273.png)

#### Expected behavior
Promises must be handled appropriately error should show in VScode.

**Comment by JoshuaKGoldberg** [comment]
Heya @Recodify - this is a known issue with the TSLint VS Code extension. It won't run "typed" rules (rules that require full information from TypeScript instead of just the basic AST information) for performance concerns. https://github.com/Microsoft/vscode-tslint/issues/70

You can try out the experimental plugin if you'd like: https://github.com/microsoft/typescript-tslint-plugin/

**Comment by Recodify** [comment]
Thanks for the quick response @JoshuaKGoldberg, Makes sense. Happy to close this.
